### PR TITLE
feature to exclude certain parts of all DOM with querySelector. 

### PR DESCRIPTION
**Pull request by JulioWar**
Is like feature of outside-if-not in  @iamadamjowett/angular-click-outside repository

### CODE REVIEW

File: src/click-outside.directive.ts [hunk 1]
```diff
@@ -1,67 +1,89 @@
...
 
-    this._document.body.removeEventListener('click', this._onClickBody);
-  }
+    private _init() {
 
-  ngOnChanges(changes: SimpleChanges) {
-    if (changes['attachOutsideOnClick'] &&
-      changes['attachOutsideOnClick'].previousValue !== changes['attachOutsideOnClick'].currentValue) {
-      this._init();
+        let nodeExcluded = this.exclude.split(',');
```
Comment (arkon) [inline src/click-outside.directive.ts]: Might be better to call this something else. I was mixing up this with the class's instance of `nodeExcluded` for a bit.

File: src/click-outside.directive.ts [hunk 2]
```diff
@@ -1,67 +1,89 @@
...
-  ElementRef,
-  EventEmitter,
-  Inject,
-  Input,
-  OnChanges,
-  OnDestroy,
-  OnInit,
-  Output,
-  SimpleChanges
+    Directive,
```
Comment (arkon) [inline src/click-outside.directive.ts]: Could we stick with 2 spaces for the indentation?

File: src/click-outside.directive.ts [hunk 3]
```diff
@@ -1,67 +1,89 @@
...
+            this._el.nativeElement.removeEventListener('click', this._initOnClickBody);
+        }
+
+        this._document.body.removeEventListener('click', this._onClickBody);
+    }
 
-  ngOnDestroy() {
-    if (this.attachOutsideOnClick) {
-      this._el.nativeElement.removeEventListener('click', this._initOnClickBody);
+    ngOnChanges(changes: SimpleChanges) {
```
Comment (arkon) [inline src/click-outside.directive.ts]: Might need to handle the case where `exclude` is updated as well.

File: src/click-outside.directive.ts [hunk 4]
```diff
@@ -1,67 +1,89 @@
...
+            }
+        }
+    }
 
-      if (this.attachOutsideOnClick) {
-        this._document.body.removeEventListener('click', this._onClickBody);
-      }
+    private isExclude(target):boolean {
+        var flag = false;
+        this.nodeExcluded.forEach((node) => {
```
Comment (arkon) [inline src/click-outside.directive.ts]: Might be better to just use a regular forloop so we can prematurely break out of it when `flag` gets set to `true`. The `return` will only break off the one iteration but still continue on for the rest.

**Review by arkon — CHANGES_REQUESTED** [review]
Thanks a lot for the PR! I've added a few comments, but they should be easy to take care of. :)

**Comment by JulioWar** [comment]
Ok, thanks for take a time to review my code, i going to see the comments and fix them.

**Comment by JulioWar** [comment]
@arkon i made the changes. see the commit.
Let me know if I have to change anything.

**Comment by arkon** [comment]
@JulioWar There appear to be merge conflicts.

**Comment by JulioWar** [comment]
@arkon but, how can i solve it? 
i dont know where exactly the conflict is and i dont have the write access to do the merge.

**Comment by arkon** [comment]
Check out this article: https://help.github.com/articles/syncing-a-fork/

**Comment by JulioWar** [comment]
Thanks for the article 👍

**Comment by arkon** [comment]
@JulioWar Thanks for the contribution! I've published v1.3.0 with it.

EDIT: make that v1.3.1. I made a stupid change before but I've fixed it since... coding while sick is a bad idea. :P

**Comment by JulioWar** [comment]
@arkon Your welcome, I learned how to make a pull request 😄

**Comment by ghost** [comment]
Can somebody please add some example on how to use exclude? What exactly is a DOM element query? I tried something like `[exclude]="button"` without success.

**Comment by JulioWar** [comment]
Hello @centigrade-thomas-becker 

You can use exclude feature like `querySelector` in pure javascript. Check [querySelector Documentation](http://www.w3schools.com/jsref/met_document_queryselector.asp). 

You need to put the value as a string, like this:
```
[exclude]="'button,.btn-primary'"
```

**Comment by arkon** [comment]
@JulioWar I added that example to the README, but it also got me wondering if it should be `querySelectorAll` in the code instead?

**Comment by JulioWar** [comment]
Yes @arkon, I agree, we can use `querySelectorAll` to get all nodes instead of use a loop

**Comment by arkon** [comment]
@JulioWar I made the change. It'll be included in the next release.

**Pull request merged**